modules/hrm/presentation/controllers/employee_controller.go

```diff
@@ -166,8 +168,62 @@ func (c *EmployeeController) Create(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if err := c.employeeService.Create(r.Context(), dto); err != nil {
-		http.Error(w, err.Error(), http.StatusInternalServerError)
+	err = c.employeeService.Create(r.Context(), dto)
+	if err != nil {
+		l, ok := composables.UseLocalizer(r.Context())
+		if !ok {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		// Check if it's a domain validation error (TIN/PIN)
+		errorsMap := map[string]string{}
+		if errors.Is(err, tax.ErrInvalidTin) {
+			tinFieldName := l.MustLocalize(&i18n.LocalizeConfig{
+				MessageID: "Employees.Private.TIN.Label",
+			})
+			errorsMap["Tin"] = l.MustLocalize(&i18n.LocalizeConfig{
+				MessageID: "ValidationErrors.custom",
+				TemplateData: map[string]string{
+					"Field": tinFieldName,
+					"Error": err.Error(),
```

Comment:
You're not localizing the error itself though. I think the custom key is redundant. The field name is obvious on the frontend. Instead you should focus on translating the error

Reply:
Also look into pkg/serrors. It was meant to be a universal error handling solution to simplify this process. Some work on the package maybe needed